### ISSUE TITLE
freedombone-prep bug

### ISSUE THREAD

**Issue by jreeve0805** [bashrc/freedombone#46]
During the initial format and copy of the SD card the /media/rootfs/etc/motd file requires write access and freedombone-prep can't write to that file. Once you remove the sd card and re-insert, hit any key to continue produces a number of error messages before the script completes, but they scroll by so fast off the page they are easy to miss. I re-formatted the sd card and tried again. This time, when the script was done copying, I removed and re-inserted the SD card, and changed the /media/rootfs/etc/motd file so it was world writable. Then when I 'hit any key to continue' the script completed without error and I got a nice motd screen with information on the bbb when I booted. I am using Ubuntu 12.04 as my host system.

**Comment by bashrc** [comment]
Some changes have been made to improve this, but the prep installer still needs more testing.

**Comment by jreeve0805** [comment]
I will test it out and get back to you.

**Comment by instance-g** [comment]
freedombome-prep not working for me too on two systems both running jessie.  In trying to fathom out what is happening, I have noticed that freedombone-prep does not make provision for systems which name sdcard disks in the form /dev/mmcblkXpX. Having made several amendments to the code installation still failed after reinserting the sdcard.

**Comment by jreeve0805** [comment]
Instance-g, I've narrowed the problem down to lines 181 - 191. These were added not long ago and it looks like the idea is to have the new jessie partitions on the SD card unmount and mount automatically. The weird part is that these commands on their own work and do what they should, but as part of the freedombone-prep script I get this error.

umount: /media/jreeve/rootfs: not mounted
umount: /media/jreeve/BOOT: mountpoint not found

It 'feels' like the unmout command is being executed before the copy of the disk image is finished. The reason I think that is because the new rootfs partition is blank. 

further testing I commented 181 - 191 out and replaced them with old code, that simply asked the user to remove the SD card, and press any button to continue. that code still works.
==== old code ===
echo ''
echo ''
read -p "Eject the microSD card, re-insert it and wait a minute for it to mount, then press any key to continue... " -n1 -s
